uiDefinition.json

```diff
@@ -18,37 +18,76 @@
-                            "request": {
-                                "method": "GET",
-                                "path": "[concat(steps('basics').scope.subscription.id, '/resourceGroups?api-version=2021-04-01')]"
-                            }
+                    {
+						"name": "scope",
+						"type": "Microsoft.Common.ResourceScope",
+						"location": {
+							"resourceTypes": []
+						}
+					},
+                    {
+                        "name": "ResourceGroupApi",
+                        "visible": false,
+                        "type": "Microsoft.Solutions.ArmApiControl",
+                        "defaultValue": "",
+                        "request": {
+                            "method": "GET",
+                            "path": "[concat(steps('basics').scope.subscription.id, '/resourceGroups?api-version=2021-04-01')]"
+                        }
+                    },
+                    {
+                        "name": "newOrExistingResourceGroup",
+                        "type": "Microsoft.Common.OptionsGroup",
+                        "label": "Use an existing resource group",
+                        "defaultValue": "false",
+                        "toolTip": "",
+                        "constraints": {
+                            "allowedValues": [
+                                {
+                                    "label": "True",
+                                    "value": true
+                                },
+                                {
+                                    "label": "False",
+                                    "value": false
+                                }
+                            ],
+                            "required": true
                         },
-                        {
-                            "name": "resourceGroup",
-                            "type": "Microsoft.Common.DropDown",
-                            "label": "Existing resource group",
-                            "multiselect": false,
-                            "defaultValue": "",
-                            "toolTip": "Select the name of the existing resource group where the image will be created. Recommended to use the same resource group where the compute gallery resides.",
-                            "filterPlaceholder": "",
-                            "defaultDescription": "",
-                            "constraints": {
-                                "allowedValues": "[map(steps('basics').resourceGroupsApi.value, (item) => parse(concat('{\"label\":\"', item.name, '\",\"value\":\"', item.name, '\"}')))]",
-                                "required": true
-                            },
-                            "infoMessages": [],
-                            "visible": true
+                        "visible": true
+                    },
+                    {
+                        "name": "existingResourceGroup",
```

Comment:
Change to "existingResourceGroupName"

Reply:
updated UI definition based on peer review